File: bestia-zoneserver/src/main/java/net/bestia/zoneserver/battle/BattleUtil.java
package net.bestia.zoneserver.battle;

final class BattleUtil {
	
	private BattleUtil() {
		// no op.
	}

	static float between(float min, float max, float val) {
		return Math.max(min, Math.min(max, val));
	}
}

File: bestia-zoneserver/src/main/kotlin/net/bestia/zoneserver/battle/BattleUtil.kt
package net.bestia.zoneserver.battle

import kotlin.math.max
import kotlin.math.min

fun Float.clamp(min: Float, max: Float = Float.MAX_VALUE): Float {
  if (this.isNaN()) {
    return min
  }
  return max(min, min(max, this))
}

fun Double.clamp(min: Double, max: Double = Double.MAX_VALUE): Double {
  if (this.isNaN()) {
    return min
  }
  return max(min, min(max, this))
}

